src/sentry/workflow_engine/processors/workflow.py

```diff
@@ -112,6 +116,27 @@ def evaluate_workflow_triggers(
             if evaluation:
                 triggered_workflows.add(workflow)
 
+    metrics.incr(
+        "workflow_engine.process_workflows.triggered_workflows",
+        len(triggered_workflows),
+        tags={"detector_type": detector.type},
+    )
+
+    environment = get_environment_by_event(event_data)
+    if environment is None:
+        return set()
```

Comment:
What if the workflow just doesn't have an environment set? That seems like it'd be ok and we don't want to exit early.

Reply:
Ah I see, yeah `Environment.DoesNotExist` sounds good